scripts/gitlab/build_and_test.sh

```diff
@@ -113,11 +105,11 @@ then
     cmake \
       -C ${hostconfig_path} \
       ${project_dir}
-    cmake --build . -j
+    cmake --build . -j 32
```

Comment:
I'm tempted to suggest a value of 12 here.
On one hand we have 7 builds happening on one quartz node and there may be more. In my experience, 12 worked even faster.
On the other hand we have only one build per lassen node.
Since it is working, I will leave this as a remark only.